### DISCUSSION TITLE
Headless templates is not running

### DISCUSSION THREAD

**Discussion by breaking153** [projectdiscovery/nuclei#2225] [Q&A] [answered]
my templates is :

```
id: basic-example1

info:
  name: Test HTTP Template
  author: pdteam
  severity: info

headless:
  - steps:
    - action: navigate
      args: 
        url: "{{BaseURL}}"
     
    matchers:
      - part: body
        type: word
        words:
          - "123456"
```
And my cmd is : .\nuclei.exe -id basic-example1 -u http://i3g8wh.dnslog.cn -dreq
![图片](https://user-images.githubusercontent.com/49428332/175779742-b791d4cc-64fb-4aaa-8abe-e5812c9faed7.png)

nuclei do not shows any error and the dnslog has no record:
![图片](https://user-images.githubusercontent.com/49428332/175779750-b0e11ae4-5f44-4d27-8668-a422ce7f7ba0.png)

when i use normal request,it will work normally:
template:
    id: basic-example1
    
    info:
      name: Test HTTP Template
      author: pdteam
      severity: info
    
    requests:
      - method: GET
        path:
          - "{{BaseURL}}"
        matchers:
          - type: word
            words:
              - "4399"
 dnslog:
![图片](https://user-images.githubusercontent.com/49428332/175779776-9d617370-a005-42d8-a94d-dd4559bc6260.png)

**Comment by breaking153** [comment]
And also,waitload cannot fix this.btw,could i use headless templates to match the fingerprints inside the js?

**Comment by ehsandeep** [comment, marked as answer]
@breaking153 to run headless templates, you need to use `-headless` CLI flag.

**Reply by breaking153** [reply]
thx a lot :D
